daml/ContingentClaims/Util.daml

```diff
@@ -80,3 +83,22 @@ pruneZeros' other = embed other
 -- | This avoids requiring the equality type constraint on `a`
 isZero Zero = True
 isZero _ = False
+
+-- | Returns the claim's implied acquisition time (IAT) if it exists.
+-- It is defined as the time `t` such that, for the purpose of lifecycling and valuation, it is irrelevant whether the contract has been acquired at `t` or at any time `s` before `t`.
+-- This check is required as currently the claim's acquisition time is not an input to `lifecycle`, which forces us to work with claims that have an IAT.
+-- For instance: 
+-- - `Scale obs c` does not have an IAT (as `obs` is evaluated at the claim's acquisition time)
+-- - the IAT for `When (TimeGte t) c` is `t`
+-- - `When (obs <= 100) c` does not have an IAT, as the acquisition time defines when we start to monitor the observable `obs`
+impliedAcquisitionTime : (Ord t) => Claim t x a -> Optional t
+impliedAcquisitionTime (Anytime (TimeGte t) _) = Some t
+impliedAcquisitionTime (When (TimeGte t) _) = Some t
+impliedAcquisitionTime (And c1 c2 cs) = foldr folder (impliedAcquisitionTime c1) (c2 :: cs)
+  where folder = flip (liftA2 min) . impliedAcquisitionTime
```

Comment:
Why the `flip`? Maybe don't worry about perf at this point and just do something like
```suggestion
min . catOptionals . fmap impliedAcquisitionTime
```

Reply:
indeed. flipping a commutative function doesn't make sense.